### PR TITLE
Change sample app to RP

### LINKED ISSUE #11794: Change sample app to a Razor Pages app
The sample app uses inline middleware code instead of MVC controllers/views or Razor Pages. So it’s not representative of the kind of code most people work with, and in several cases snippets have to be inline in the Markdown because the code wouldn’t be representative of MVC/RP code.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0e03fdf4-b19b-17c3-04f4-336ab6d2ee06
* Version Independent ID: 38515dfb-91a5-b395-db9d-084bbaf095c8
* Content: [Handle errors in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-2.1#additional-resources)
* Content Source: [aspnetcore/fundamentals/error-handling.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/error-handling.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### LINKED ISSUE #11859: High traffic low csat
Only 25% of raters mark the article as helpful.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0e03fdf4-b19b-17c3-04f4-336ab6d2ee06
* Version Independent ID: 38515dfb-91a5-b395-db9d-084bbaf095c8
* Content: [Handle errors in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-2.1#additional-resources)
* Content Source: [aspnetcore/fundamentals/error-handling.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/error-handling.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #11794 
Fixes #11859 (no verbatim comments available to explain why so many "not helpful" votes, so we're changing the sample app to RP plus a general edit for clarity and will recheck CSAT later to find out if that improved the rating.

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-2.1&branch=pr-en-us-11899)

### CODE REVIEW

File: aspnetcore/fundamentals/error-handling.md [hunk 1]
```diff
@@ -14,41 +14,40 @@ By [Tom Dykstra](https://github.com/tdykstra/), [Luke Latham](https://github.com
...
-To see the Developer Exception Page, run the sample app with the environment set to `Development` and add `?throw=true` to the base URL of the app. The page includes the following information about the exception and the request:
+The page includes the following information about the exception and the request:
 
 * Stack trace
 * Query string parameters (if any)
 * Cookies (if any)
 * Headers
 
-## Configure a custom exception handling page
+To see the Developer Exception Page in the sample app, use the `DevEnvironment` preprocessor directive and select **Trigger an exception** on the home page.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/error-handling.md]: To see the Developer Exception Page in the [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/error-handling/samples/2.x), use the `DevEnvironment` preprocessor directive and select **Trigger an exception** on the home page.

File: aspnetcore/fundamentals/error-handling.md [hunk 2]
```diff
@@ -61,49 +60,39 @@ public IActionResult Error()
...
-By default, an ASP.NET Core app doesn't provide a status code page for HTTP status codes, such as *404 - Not Found*. The app returns a status code and an empty response body. To provide status code pages, use Status Code Pages Middleware.
+## UseStatusCodePages
 
-The middleware is made available by the [Microsoft.AspNetCore.Diagnostics](https://www.nuget.org/packages/Microsoft.AspNetCore.Diagnostics/) package, which is available in the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).
+By default, an ASP.NET Core app doesn't provide a status code page for HTTP status codes, such as *404 - Not Found*. The app returns a status code and an empty response body. To provide status code pages, use Status Code Pages middleware.
 
-### UseStatusCodePages
+The middleware is made available by the [Microsoft.AspNetCore.Diagnostics](https://www.nuget.org/packages/Microsoft.AspNetCore.Diagnostics/) package, which is in the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).
 
 To enable default text-only handlers for common error status codes, add the following code to the `Startup.Configure` method:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/error-handling.md]: To enable default text-only handlers for common error status codes, add <xref:Microsoft.AspNetCore.Builder.StatusCodePagesExtensions.UseStatusCodePages*> to `Startup.Configure`:

File: aspnetcore/fundamentals/error-handling.md [hunk 3]
```diff
@@ -113,70 +102,63 @@ Here's an example of text displayed by the default handlers:
 Status Code: 404; Not Found
 ```
 
-The middleware supports several extension methods that allow you to customize its behavior.
+## UseStatusCodePages with format string
 
-### UseStatusCodePages with lambda
+An overload of <xref:Microsoft.AspNetCore.Builder.StatusCodePagesExtensions.UseStatusCodePages*> takes a content type and format string, which you can use to customize the content type and response text:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/error-handling.md]: An overload of <xref:Microsoft.AspNetCore.Builder.StatusCodePagesExtensions.UseStatusCodePages*> takes a content type and format string, can be used to customize the content type and response text:

File: aspnetcore/fundamentals/error-handling.md [hunk 4]
```diff
@@ -113,70 +102,63 @@ Here's an example of text displayed by the default handlers:
...
-<xref:Microsoft.AspNetCore.Builder.StatusCodePagesExtensions.UseStatusCodePagesWithRedirects*>:
+The <xref:Microsoft.AspNetCore.Builder.StatusCodePagesExtensions.UseStatusCodePagesWithRedirects*> extension method:
 
 * Sends a *302 - Found* status code to the client.
 * Redirects the client to the location provided in the URL template.
 
 [!code-csharp[](error-handling/samples/2.x/ErrorHandlingSample/Startup.cs?name=snippet_StatusCodePagesWithRedirect)]
 
-<xref:Microsoft.AspNetCore.Builder.StatusCodePagesExtensions.UseStatusCodePagesWithRedirects*> is commonly used when the app:
+The URL template can include a `{0}` placeholder for the status code, as shown in the example. If the URL template starts with a tilde (~), the tilde is replaced by the app's `PathBase`. If you point to an endpoint within the app, create an MVC view or Razor page for the endpoint. For a Razor Pages example, see [StatusCode.cshtml](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/error-handling/samples/2.x/Pages/StatusCode.cshtml) in the sample app.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/error-handling.md]: The URL template can include a `{0}` placeholder for the status code, as shown in the example. If the URL template starts with a tilde (~), the tilde is replaced by the app's `PathBase`. If an endpoint is used in the app, create an MVC view or Razor page for the endpoint. For a Razor Pages example, see [StatusCode.cshtml](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/error-handling/samples/2.x/Pages/StatusCode.cshtml) in the sample app.

**Comment by Rick-Anderson** [comment]
>Also, be aware that once the headers for a response are sent:
Working on review, it will be done soon. Also is a stevism that's banned.

once the headers for a response are sent:
or
**Note**: once the headers for a response are sent:

File: aspnetcore/fundamentals/error-handling.md [hunk 5]
```diff
@@ -310,36 +180,41 @@ Also, be aware that once the headers for a response are sent:
...
 
 ## Startup exception handling
 
-Only the hosting layer can handle exceptions that take place during app startup. Using [Web Host](xref:fundamentals/host/web-host), you can [configure how the host behaves in response to errors during startup](xref:fundamentals/host/web-host#detailed-errors) with the `captureStartupErrors` and `detailedErrors` keys.
+Only the hosting layer can handle exceptions that take place during app startup. Using [Web Host](xref:fundamentals/host/web-host), you can [configure how the host behaves in response to errors during startup](xref:fundamentals/host/web-host#detailed-errors).
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/error-handling.md]: Only the hosting layer can handle exceptions that take place during application startup. Using [Web Host](xref:fundamentals/host/web-host), you can [configure how the host behaves in response to errors during startup](xref:fundamentals/host/web-host#detailed-errors).

**Comment by Rick-Anderson** [comment]
@guardrex  see 
##Server exception handling
Can you recommend a link to debug  ` 500 - Internal Server Error`

File: aspnetcore/fundamentals/error-handling.md [hunk 6]
```diff
@@ -310,36 +180,41 @@ Also, be aware that once the headers for a response are sent:
...
+{
+    app.UseDatabaseErrorPage();
+}
+```
 
-### Exception filters
+## Exception filters
 
-Exception filters can be configured globally or on a per-controller or per-action basis in an MVC app. These filters handle any unhandled exception that occurs during the execution of a controller action or another filter. These filters aren't called otherwise. For more information, see <xref:mvc/controllers/filters#exception-filters>.
+Exception filters can be configured globally or on a per-controller or per-action basis in an MVC app. These filters handle any unhandled exception that occurs during the execution of a controller action or another filter. For more information, see <xref:mvc/controllers/filters#exception-filters>.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/error-handling.md]: What about RP?

**Comment by guardrex** [comment]
> Can you recommend a link to debug 500 - Internal Server Error

Not too well in this case because it refers to server error handling outside of the framework. We do have IIS and Azure coverage, but we don't have Linux and other servers covered.

If you want the Azure and IIS ones, they're at ...

* \<xref:host-and-deploy/azure-apps/troubleshoot>
* \<xref:host-and-deploy/iis/troubleshoot>

File: aspnetcore/fundamentals/error-handling.md [hunk 7]
```diff
@@ -34,7 +34,8 @@ The page includes the following information about the exception and the request:
 * Cookies (if any)
 * Headers
 
-To see the Developer Exception Page in the sample app, use the `DevEnvironment` preprocessor directive and select **Trigger an exception** on the home page.
+To see the Developer Exception Page in the 
+[sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/error-handling/samples/2.x), use the `DevEnvironment` preprocessor directive and select **Trigger an exception** on the home page.
```
Comment (guardrex) [inline aspnetcore/fundamentals/error-handling.md]: fyi ... I always link to the *samples* folder and let the reader pick the version. No need to update the links that way going forward.

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/error-handling.md]: That's what I do too.

**Pull request merged**